src/support/HttpConsumer.cpp

```diff
@@ -10,9 +10,12 @@ Vault::HttpConsumer::get(const Vault::Client &client, const Vault::Url &url) {
 
   auto response = client.getHttpClient().get(url, client.getToken(), client.getNamespace());
 
-  return HttpClient::is_success(response)
-         ? std::optional<std::string>(response.value().body.value())
-         : std::nullopt;
+  if (HttpClient::is_success(response)) {
+    return std::optional<std::string>(response.value().body.value());
+  } else {
+    client.getHttpClient().reportError(response);
```

Comment:
In all of these cases, we should replace the report error with the fungible `responseErrorCallback`.

Reply:
This allows the consumer to inject json parsing if they choose, but it doesn't demand it.